hack/generator/pkg/astmodel/conversion_function_builder.go

```diff
@@ -0,0 +1,494 @@
+
+	destinationType, ok := params.DestinationType.(*ArrayType)
+	if !ok {
+		return nil
+	}
+
+	sourceType, ok := params.SourceType.(*ArrayType)
+	if !ok {
+		return nil
+	}
+
+	var results []dst.Stmt
+
+	itemIdent := "item"
+	elemIdent := "elem"
+
+	depth := params.CountArraysAndMapsInConversionContext()
+
+	elemType := destinationType.Element()
+	actualDestination := params.ClonedDestination() // TODO: improve name
+	if depth > 0 {
+		actualDestination = dst.NewIdent(elemIdent)
+		results = append(
+			results,
+			astbuilder.LocalVariableDeclaration(
+				elemIdent,
+				destinationType.AsType(builder.CodeGenerationContext),
+				""))
+		elemIdent = fmt.Sprintf("elem%d", depth)
+	}
+
+	result := &dst.RangeStmt{
+		Key:   dst.NewIdent("_"),
+		Value: dst.NewIdent(itemIdent),
+		X:     params.ClonedSource(),
+		Tok:   token.DEFINE,
+		Body: &dst.BlockStmt{
+			List: builder.BuildConversion(
+				ConversionParameters{
+					Source:            dst.NewIdent(itemIdent),
+					SourceType:        sourceType.Element(),
+					Destination:       dst.Clone(actualDestination).(dst.Expr),
+					DestinationType:   elemType,
+					NameHint:          elemIdent,
+					ConversionContext: append(params.ConversionContext, destinationType),
+					AssignmentHandler: astbuilder.AppendList,
+				}),
+		},
+	}
+	results = append(results, result)
+
+	// If we have an assignment handler, we need to make sure to call it. This only happens in the case of nested
+	// maps/arrays, where we need to make sure we generate the map assignment/array append before returning (otherwise
+	// the "actual" assignment will just end up being to an empty array/map).
+	if params.AssignmentHandler != nil {
+		results = append(results, params.AssignmentHandler(params.ClonedDestination(), dst.Clone(actualDestination).(dst.Expr)))
+	}
+
+	return results
+}
+
+// IdentityConvertComplexMapProperty handles conversion for map properties with complex values.
+// This function panics if the map keys are not primitive types.
+// This function generates code that looks like this:
+// 	if <source> != nil {
+//		<destination> = make(map[<destinationType.KeyType()]<destinationType.ValueType()>)
+//		for key, value := range <source> {
+// 			<code for producing result from destinationType.ValueType()>
+//			<destination>[key] = <result>
+//		}
+//	}
+func IdentityConvertComplexMapProperty(builder *ConversionFunctionBuilder, params ConversionParameters) []dst.Stmt {
+	destinationType, ok := params.DestinationType.(*MapType)
+	if !ok {
+		return nil
+	}
+
+	sourceType, ok := params.SourceType.(*MapType)
+	if !ok {
+		return nil
+	}
+
+	if _, ok := destinationType.KeyType().(*PrimitiveType); !ok {
+		panic(fmt.Sprintf("map had non-primitive key type: %v", destinationType.KeyType()))
+	}
+
+	depth := params.CountArraysAndMapsInConversionContext()
+
+	keyIdent := "key"
+	valueIdent := "value"
+	elemIdent := "elem"
```

Comment:
Maybe `item`?

Reply:
Or if this is the list, how about taking the singular form of the property name and suffixing `List` ?

Reply:
Same comment as above about holding off on generated name changes until a second PR to ease review burden.